packages/editor-ui/src/plugins/i18n/index.ts

```diff
@@ -302,14 +363,53 @@ export async function loadLanguage(language?: string) {
 	setLanguage(language);
 }
 
+/**
+ * Add a node translation to the i18n instance's `messages` object.
+ */
 export function addNodeTranslation(
-	nodeTranslation: { [key: string]: object },
+	nodeTranslation: { [nodeType: string]: object },
+	language: string,
+) {
+	const oldNodesBase = i18nInstance.messages[language]['n8n-nodes-base'] || {};
+
+	const updatedNodes = {
+		// @ts-ignore
+		...oldNodesBase.nodes,
+		...nodeTranslation,
+	};
+
+	const newNodesBase = {
+		'n8n-nodes-base': Object.assign(
+			oldNodesBase,
+			{ nodes: updatedNodes },
+		),
+	};
+
+	i18nInstance.setLocaleMessage(
+		language,
+		Object.assign(i18nInstance.messages[language], newNodesBase),
+	);
+}
+
+/**
+ * Add a credential translation to the i18n instance's `messages` object.
+ */
+export function addCredentialTranslation(
+	nodeCredentialTranslation: { [credentialType: string]: object },
 	language: string,
 ) {
+	const oldNodesBase = i18nInstance.messages[language]['n8n-nodes-base'] || {};
+
+	const updatedCredentials = {
+		// @ts-ignore
+		...oldNodesBase.credentials,
+		...nodeCredentialTranslation,
+	};
+
 	const newNodesBase = {
 		'n8n-nodes-base': Object.assign(
-			i18nInstance.messages[language]['n8n-nodes-base'] || {},
-			nodeTranslation,
+			oldNodesBase,
+			{ credentials: updatedCredentials },
 		),
```

Comment:
reference previous comment

Reply:
Suggested edit working for some reason. Will revisit later to avoid delaying the release.